src/python/test/test_dxclient.py

```diff
@@ -3248,21 +3248,94 @@ def test_dx_run_extra_args(self):
         with self.assertSubprocessFailure(stderr_regexp='JSON', exit_code=3):
             run("dx run " + applet_id + " --extra-args not-a-JSON-string")
 
+    def test_dx_run_sys_reqs(self):
+        app_spec = {"project": self.project,
+                    "dxapi": "1.0.0",
+                    "runSpec": {"interpreter": "bash",
+                                "distribution": "Ubuntu",
+                                "release": "20.04",
+                                "version": "0",
+                                "code": "echo 'hello'",
+                                "systemRequirements": {
+                                    "main": {
+                                        "instanceType": "mem2_hdd2_x1",
+                                        "clusterSpec": {"type": "spark",
+                                                        "initialInstanceCount": 1,
+                                                        "version": "2.4.4",
+                                                        "bootstrapScript": "x.sh"}
+                                    },
+                                    "other": {
+                                        "instanceType": "mem2_hdd2_x4",
+                                        "clusterSpec": {"type": "spark",
+                                                        "initialInstanceCount": 5,
+                                                        "version": "2.4.4",
+                                                        "bootstrapScript": "x.sh"}
+                                    },
+                                }
+                                }
+                    }
+        applet_id = dxpy.api.applet_new(app_spec)['id']
+        requested_inst_type_by_exec = {
+            applet_id: {
+                "*": {
+                    "instanceType": "mem2_ssd1_v2_x2",
+                    "clusterSpec": {"initialInstanceCount": 3}}}}
+
+        (stdout, stderr) = run('_DX_DEBUG=2 dx run ' + applet_id + ' ' +
+                               '--instance-type mem2_hdd2_x2 ' +
+                               '--instance-count 15 ' +
+                               '--instance-type-by-executable ' +
+                               '\'' +
+                               json.dumps(requested_inst_type_by_exec) + '\'',
+                               also_return_stderr=True)
+        expected_sys_reqs_by_exec = '"systemRequirementsByExecutable": ' + \
+            json.dumps(requested_inst_type_by_exec)
+        self.assertIn(expected_sys_reqs_by_exec, stderr)
```

Comment:
@commandlinegirl  I was using  _DX_DEBUG to validate the `systemRequirementsByExecutable` in api request